### DISCUSSION TITLE
The final results after switching to AtlantaFX

### DISCUSSION THREAD

**Discussion by crschnick** [mkpaz/atlantafx#43] [General]
I just wanted to let you know that over the last few weeks I switched my project [xpipe](https://github.com/xpipe-io/xpipe) over to completely use AtlantaFX and everything looks and works great!

I adapted your file browser example to work on and across arbitrary remote servers through shell connections:

![image](https://user-images.githubusercontent.com/72509152/221438071-e3f287ea-a6ee-49b3-a776-4b5b39b1fd92.png)

I also altered the default PreferencesFX theme to look better:

![image](https://user-images.githubusercontent.com/72509152/221438186-aa36e502-3e05-4e71-b8cd-52d92e0ddd1a.png)

The connection manager also looks much clearer now:

![image](https://user-images.githubusercontent.com/72509152/221438232-18bc1ee3-4b3f-49dc-8fb0-df8896aa7c91.png)

**Comment by mkpaz** [comment]
Thanks for sharing! It looks pretty nice. Next version I plan a couple of changes to simplify creating preferences UI.

**Comment by CamilYed** [comment]
Wow :) 


Wow, your application looks amazing.  
However, I noticed that virtual threads don’t seem to work on Windows and ARM64.

Do you have any tips for a beginner on how to use Atlanta FX?  
I have a setup like this and I’m wondering if it’s enough to start with.


```java

plugins {
    id("application")
    id("org.openjfx.javafxplugin") version "0.1.0"
    id("org.beryx.jlink") version "3.1.1"
    id("jacoco")
    id("org.sonarqube") version "7.2.2.6593"
}

val javaVersion: String by extra { "25" }
val javafxVersion: String by extra { "25" }
val atlantaFxVersion: String by extra { "2.0.1" }
val jacksonVersion: String by extra { "2.18.2" }
val junitVersion: String by extra { "5.11.4" }
val assertjVersion: String by extra { "3.27.3" }
val wireMockVersion: String by extra { "3.13.2" }
val testFxVersion: String by extra { "4.0.18" }
val hamcrestVersion: String by extra { "3.0" }
val ikonliVersion: String by extra { "12.4.0" }
val junitLauncherVersion: String by extra { "1.11.4" }

group = "com.camilyed.jbolt"
version = "1.0.0-SNAPSHOT"

repositories {
    mavenCentral()
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(javaVersion))
    }
}

javafx {
    version = javafxVersion
    modules("javafx.controls", "javafx.fxml", "javafx.graphics")
}

application {
    mainModule.set("github.com.camilyed.jbolt")
    mainClass.set("github.com.camilyed.jbolt.App")
}

dependencies {
    implementation("io.github.mkpaz:atlantafx-base:$atlantaFxVersion")
    implementation("com.fasterxml.jackson.core:jackson-databind:$jacksonVersion")
    implementation("org.kordamp.ikonli:ikonli-javafx:${ikonliVersion}")
    implementation("org.kordamp.ikonli:ikonli-materialdesign2-pack:${ikonliVersion}")

    testImplementation(platform("org.junit:junit-bom:$junitVersion"))
    testImplementation("org.junit.jupiter:junit-jupiter")
    testImplementation("org.assertj:assertj-core:$assertjVersion")
    testImplementation("org.wiremock:wiremock:$wireMockVersion")
    testImplementation("org.hamcrest:hamcrest:${hamcrestVersion}")
    testImplementation("org.testfx:testfx-core:$testFxVersion")
    testImplementation("org.testfx:testfx-junit5:$testFxVersion")
    testRuntimeOnly("org.junit.platform:junit-platform-launcher:${junitLauncherVersion}")
    testRuntimeOnly("org.openjfx:javafx-swing:$javafxVersion")
}

sonar {
    properties {
        property("sonar.projectKey", "SoftwareJCompany_Jbolt")
        property("sonar.organization", "softwarejcompany")
        property("sonar.host.url", "https://sonarcloud.io")
        property("sonar.coverage.jacoco.xmlReportPaths", "build/reports/jacoco/test/jacocoTestReport.xml")
        property("sonar.exclusions", "**/App.java")
    }
}

tasks.test {
    useJUnitPlatform()
    testLogging {
        events("passed", "skipped", "failed")
    }
    include("**/*Test.class", "**/*IT.class")

    systemProperty("java.awt.headless", "false")
    systemProperty("testfx.robot", "glass")
    systemProperty("testfx.headless", "false")
    systemProperty("prism.order", "sw")
}

tasks.jacocoTestReport {
    dependsOn(tasks.test)

    reports {
        xml.required.set(true)
        html.required.set(true)
    }
}

tasks.test {
    finalizedBy(tasks.jacocoTestReport)
}

```

**Comment by crschnick** [comment]
@CamilYed If you are looking for a good place to start, I actually built a template based off the xpipe application that is built around AtlantaFX: https://github.com/xpipe-io/kickstartfx. And yes, the virtual threads are disabled because that caused some crashes on ARM64 in Liberica (https://github.com/bell-sw/Liberica/issues/211)

**Reply by CamilYed** [reply]
This looks impressive! Great job on KickstartFX.
I have a quick question regarding the development workflow: Does this project integrate well with FXML files, or is it designed to be used primarily with a programmatic UI approach? Also, did you use Scene Builder for designing the layouts, or was it all coded manually?
I'm considering using it for a high-performance tool, so understanding the UI structure would be very helpful. Thanks!

**Reply by crschnick** [reply]
If you want to get serious, you shoudn't be using fxml anymore. It's acceptable to get started with it, but it just lacks the depth for designing complex UIs.

So kickstartfx does not use any fxml, everything is created manually
